indicators/NO_WECO_001-012/R/NO_WECO_001-012.qmd

```diff
@@ -0,0 +1,937 @@
+-->
+
+
+## 9. Analyses
+
+### 9.1. Preparing and testing the data
+
+We start by loading some functions and data from the [WFD2ECA](https://github.com/NINAnor/WFD2ECA/) repository ([Sandvik 2025](#san25)), which will be used in the following analyses.
+
+```{r}
+source(url("https://github.com/NINAnor/WFD2ECA/raw/refs/heads/main/R/Funksjon.R" ))
+load(  url("https://github.com/NINAnor/WFD2ECA/raw/refs/heads/main/data/VF.RData"))
+```
+
+In addition, some specialised functions need to be defined for translating ecological status or potential scores into nEQRs, and for creating randomising nEQR values under different uncertainty conditions.
+
+```{r}
+# Function returning the midpoint nEQR of each ecological status category
+status2nEQR <- function(status) return(
+  ifelse(status == "Svært god",    0.9,
+  ifelse(status == "God",          0.7,
+  ifelse(status == "Moderat",      0.5,
+  ifelse(status == "Dårlig",       0.3,
+  ifelse(status == "Svært dårlig", 0.1,
+  NA)))))
+)
+
+
+
+# Function returning the midpoint nEQR of each ecological potential category
+potential2nEQR <- function(potential) return(
+  ifelse(potential == "Godt",         0.525,
+  ifelse(potential == "Moderat",      0.375,
+  ifelse(potential == "Dårlig",       0.225,
+  ifelse(potential == "Svært dårlig", 0.075,
+  NA))))
+)
+
+
+
+# Function creating random nEQR values for estimates of low uncertainty
+lowUncertainty <- function(m, pot = FALSE) {
+  L <- length(m)
+  U <- if (pot) 0.6 else 1
+  D <- if (pot) 0.075 else 0.1
+  m[] <- m[] + runif(L, -D, +D)
+  return(sapply(sapply(m[], max, 0), min, U))
+}
+
+
+
+# Function creating random nEQR values for estimates of indermediate uncertainty
+intermediateUncertainty <- function(m, pot = FALSE) {
+  L <- length(m)
+  U <- if (pot) 0.6 else 1
+  D <- if (pot) 0.15 else 0.2
+  R <- runif(L)
+  m <- m - D/2
+  m[] <- ifelse(R < 0.125, m[]     - rbeta(L, 1, 2) * D / 3,
+         ifelse(R > 0.875, m[] + D + rbeta(L, 1, 2) * D / 3,
+                           m[]     + runif(L, 0, 1) * D))
+  return(sapply(sapply(m, max, 0), min, U))
+}
+
+
+
+# Function creating random nEQR values for estimates of high uncertainty
+highUncertainty <- function(m, pot = FALSE) {
+  L <- length(m)
+  U <- if (pot) 0.6 else 1
+  D <- if (pot) 0.15 else 0.2
+  R <- runif(L)
+  m <- m - D/2
+  m[] <- ifelse(R < 0.25, m[]     - rbeta(L, 1, 2) * D,
+         ifelse(R > 0.75, m[] + D + rbeta(L, 1, 2) * D,
+                          m[]     + runif(L, 0, 1) * D))
+  return(sapply(sapply(m[], max, 0), min, U))
+}
+```
+
+We define the quantiles we want to estimate for each indicator as the median, the quartiles and the lower/upper 95% confidence levels.
+
+```{r}
+Qnt <- c(0.025, 0.25, 0.5, 0.75, 0.975)
+```
+
+The data from *Vann-nett* need to be prepared for analysis.
+
+```{r}
+# Column names are changed so that they match the ones used in the WFD2ECA repo:
+names(V) <- c("id", "nam", "kat", "areal", "nas", "smvf", 
```

Comment:
This can be made more transparent and safe by using a new name - old name syntax, lik ein dplyr::rename(). Just something to consider for next time.